src/map/script.c

```diff
@@ -12915,39 +12949,42 @@ static BUILDIN(getwaitingroomstate)
 /// The id's of the teleported players are put into the array $@warpwaitingpc[]
 /// The total number of teleported players is put into $@warpwaitingpcnum
 ///
-/// warpwaitingpc "<map name>",<x>,<y>,<number of players>;
+/// warpwaitingpc "<map name>",<x>,<y>,<number of players>{,<npcname>};
 /// warpwaitingpc "<map name>",<x>,<y>;
 static BUILDIN(warpwaitingpc)
 {
-	int x, y, i, n;
-	const char* map_name;
-	struct npc_data* nd;
-	struct chat_data* cd;
+	int i, n;
+	const char *map_name = script_getstr(st, 2);
+	int x = script_getnum(st, 3);
+	int y = script_getnum(st, 4);
+	struct npc_data *nd;
+	struct chat_data *cd;
 
-	nd = map->id2nd(st->oid);
-	if (nd == NULL || (cd=map->id2cd(nd->chat_id)) == NULL)
+	if (script_hasdata(st, 6))
+		nd = npc->name2id(script_getstr(st, 6));
+	else
+		nd = map->id2nd(st->oid);
+
+	if (nd == NULL || (cd = map->id2cd(nd->chat_id)) == NULL)
 		return true;
```

Comment:
should throw warning npc not found and return false;

isn't that you have done it in your `*waitingroomkick` script command ?
`ShowWarning("buildin_waitingroomkick: NPC name '%s' not found.\n", script_getstr(st, 2));`

Reply:
added

Reply:
```c
	struct npc_data *nd;
	if (script_hasdata(st, 6)) {
		nd = npc->name2id(script_getstr(st, 6));
		if (nd == NULL) {
			ShowWarning("buildin_warpwaitingpc: NPC '%s' not found.\n", script_getstr(st, 6));
			return false;
		}
	} else {
		nd = map->id2nd(st->oid);
		if (nd == NULL) {
			ShowWarning("buildin_warpwaitingpc: NPC not found.\n");
			return false;
		}
	}

	struct chat_data *cd = map->id2cd(nd->chat_id);
	if (cd == NULL) {
		ShowWarning("buildin_warpwaitingpc: NPC '%s' does not have a chatroom.\n", nd->name);
		return false;
	}
```